src/instance.rs

```diff
@@ -1254,7 +1254,7 @@ impl<T> Py<T> {
     where
         D: FromPyObject<'py>,
     {
-        FromPyObject::extract(unsafe { py.from_borrowed_ptr(self.as_ptr()) })
+        FromPyObject::extract_bound(unsafe { &Bound::from_borrowed_ptr(py, self.as_ptr()) })
```

Comment:
Maybe we can do this using safe wrappers?

```suggestion
        self.bind(py).extract()
```